internal/translation/datafederation/conversion_endpoints.go

```diff
@@ -3,58 +3,76 @@ package datafederation
 import (
 	"fmt"
 
+	"github.com/mongodb/mongodb-atlas-kubernetes/v2/internal/pointer"
+
 	"go.mongodb.org/atlas-sdk/v20231115008/admin"
 
 	"github.com/mongodb/mongodb-atlas-kubernetes/v2/internal/cmp"
-	"github.com/mongodb/mongodb-atlas-kubernetes/v2/internal/pointer"
 	akov2 "github.com/mongodb/mongodb-atlas-kubernetes/v2/pkg/api/v1"
 )
 
-type DatafederationPrivateEndpointEntry struct {
+type PrivateEndpoint struct {
 	*akov2.DataFederationPE
 	ProjectID string
 }
 
-func NewDatafederationPrivateEndpointEntry(pe *akov2.DataFederationPE, projectID string) *DatafederationPrivateEndpointEntry {
+func NewPrivateEndpoints(projectID string, privateEndpoints []akov2.DataFederationPE) ([]PrivateEndpoint, error) {
+	entries := make([]PrivateEndpoint, 0, len(privateEndpoints))
+	for _, privateEndpoint := range privateEndpoints {
+		entries = append(entries, NewPrivateEndpoint(projectID, privateEndpoint.DeepCopy()))
+	}
+
+	if err := cmp.Normalize(entries); err != nil {
```

Comment:
I believe adding a map that considers the endpoint ID is more formally correct, see https://github.com/mongodb/mongodb-atlas-kubernetes/pull/1929/files#diff-11a090be55936627141c551ed7808292f8dc76db3baece47ad78ad24fb89d730R73.